spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -20,26 +20,52 @@ def foobar_action
 end
 
 RSpec.describe Hq::PartnersController, type: :controller do
-  before :each do
-    sign_in instance_double(AdminUser)
-    @partner = build_stubbed :partner
-    expect(Partner).to receive(:find).and_return(@partner)
-  end
 
-  specify 'editing renders the edit view' do
-    get :edit, id: @partner.id
-    expect(response).to render_template 'edit'
-  end
+  context '#new and #create' do
+    
+    before :each do
+      sign_in instance_double(AdminUser)
+      @partner = FactoryGirl.build(:partner)
+    end
+    
+    specify 'successful create redirects to the show view' do
+      post :create, partner: @partner.attributes
+      @partner = Partner.find_by_name(@partner[:name])
+      expect(response).to redirect_to hq_partner_path(@partner)
+    end
```

Comment:
Ok, this re-write will work:

```ruby
    specify 'successful create redirects to the show view', :focus => true do
      @partner = double Partner, :id => 1
      allow(@partner).to receive(:save).and_return(@partner)
      post :create, partner: {name: 'partner', province_id: Province.first.id, start_date: Date.current}
      expect(response).to redirect_to hq_partner_path(1)
  end
```

Still touching the database though (when looking for a Province). Otherwise, `@partner.save` will fail validation. Is there any way to stub the province out as well?